### PR TITLE
A Lighter Timer

### PR DESCRIPTION
**Pull request by donglihe-hub**
## What does this PR do?

In many situations, developers may want to measure the running time of functions. The original Timer in LibMultiLabel is no more convenient than using the time package directly. The new lightweight timer is a decorator and is easy to use.

usage 1:
```python
import time
from libmultilabel.common_utils import timer

@timer
def a ():
    time.sleep(3)

a()
```
```
INFO:lightning.pytorch.utilities.rank_zero:Finished 'a' in 3.00 seconds
```
usage 2:
```python
def a ():
    time.sleep(3)

timer(a)()
```
```
INFO:lightning.pytorch.utilities.rank_zero:Finished 'a' in 3.00 seconds
```

## Test CLI & API (`bash tests/autotest.sh`)
Test APIs used by main.py.
- [x] Test Pass
  - (Copy and paste the last outputted line here.)
- [ ] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
If any new APIs are added, please check if the description of the APIs is added to API document. 
- [ ] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [ ] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_changed_document.sh`)
If any APIs in quickstarts or tutorials are modified, please run this test to check if the current examples can run correctly after the modified APIs are released.

### CODE REVIEW

**Comment by Eleven1Liu** [comment]
@Sinacam Review

File: libmultilabel/common_utils.py [hunk 1]
```diff
@@ -156,3 +127,17 @@ def is_multiclass_dataset(dataset, label="label"):
             a multi-class problem."""
         )
     return ratio == 1.0
+
+
+def timer(func):
+    """Emit info-level wall time only on global rank 0."""
```
Comment (Eleven1Liu) [inline libmultilabel/common_utils.py:133]: Why global rank 0 here?

**Review by Eleven1Liu — CHANGES_REQUESTED** [review]
Please update the comment.

File: libmultilabel/common_utils.py (hunk 1 above)
Comment (donglihe-hub) [reply libmultilabel/common_utils.py]: Removed

**Review by Eleven1Liu — APPROVED** [review]

**Pull request merged**